llvm/include/llvm/IR/IntrinsicsSPIRV.td

```diff
@@ -20,7 +20,7 @@ let TargetPrefix = "spv" in {
 
   def int_spv_gep : Intrinsic<[llvm_anyptr_ty], [llvm_i1_ty, llvm_any_ty, llvm_vararg_ty], [ImmArg<ArgIndex<0>>]>;
   def int_spv_load : Intrinsic<[llvm_i32_ty], [llvm_anyptr_ty, llvm_i16_ty, llvm_i8_ty], [ImmArg<ArgIndex<1>>, ImmArg<ArgIndex<2>>]>;
-  def int_spv_store : Intrinsic<[], [llvm_i32_ty, llvm_anyptr_ty, llvm_i16_ty, llvm_i8_ty], [ImmArg<ArgIndex<2>>, ImmArg<ArgIndex<3>>]>;
+  def int_spv_store : Intrinsic<[], [llvm_any_ty, llvm_anyptr_ty, llvm_i16_ty, llvm_i8_ty], [ImmArg<ArgIndex<2>>, ImmArg<ArgIndex<3>>]>;
```

Comment:
Could you pls provide an example requiring these store-related changes? `nested-composites.ll` doesn't need them, so maybe it's even worth to add a new test to better illustrate the changes?

Reply:
There are about 400 compilation fails in current CTS *lls shared by Michal. Most of them are related to the issue with StoreInst. There are some of them (one for each affected test):
```
math_brute_force/modf/OCL_asm83205908d2d9b9fc_before_spirv_backend.ll
math_brute_force/fract/OCL_asmb2ad4538ae54823b_before_spirv_backend.ll
math_brute_force/remquo/OCL_asm15e90ce8357c7bed_before_spirv_backend.ll
math_brute_force/lgamma_r/OCL_asm5c822751931b8422_before_spirv_backend.ll
math_brute_force/sincos/OCL_asmce80d25172356ece_before_spirv_backend.ll
generic_address_space/generic_variable_gentype/OCL_asmfb1db4d11def20a0_before_spirv_backend.ll
relationals/shuffle_copy/OCL_asm70ee5d1be466faa0_before_spirv_backend.ll
relationals/shuffle_function_call/OCL_asm342bf149d7bfba85_before_spirv_backend.ll
relationals/shuffle_array_cast/OCL_asm420b7e961b3defd2_before_spirv_backend.ll
basic/progvar_prog_scope_uninit/OCL_asm24bf055555a90507_before_spirv_backend.ll
basic/vstore_private/OCL_asmf4d78f163d3a5a7d_before_spirv_backend.ll
```
After the fix, there are only 19 fails.

>  so maybe it's even worth to add a new test to better illustrate the changes?
It's a good idea, I will make a simplified test out of one of the above.

Reply:
> There are about 400 compilation fails in current CTS *lls shared by Michal. Most of them are related to the issue with StoreInst. There are some of them (one for each affected test):

What's the issue in particular, could you pls illustrate it with some IR example?

Reply:
Sure. It's verifier issue (before IR translation):
```
>bin/llc -O0 /storage/home/idyachkov/spirv/OpenCL_CTS_2021/basic/vstore_private/OCL_asmf4d78f163d3a5a7d_before_spirv_backend.ll -o e1.s
Stored value type does not match pointer operand type!
  store i32 %11, <4 x i16>* %storetmp, align 8
 i32in function test_fn
LLVM ERROR: Broken function found, compilation aborted!
```
The patch inserts intrinsic call instead of the StoreInst